libsolidity/interface/CompilerStack.cpp

```diff
@@ -173,6 +173,15 @@ bool CompilerStack::parse()
 				if (!resolver.updateDeclaration(*m_globalContext->currentThis())) return false;
 				if (!resolver.updateDeclaration(*m_globalContext->currentSuper())) return false;
 				if (!resolver.resolveNamesAndTypes(*contract)) return false;
+				if (m_contracts.find(contract->name()) != m_contracts.end())
+				{
+					const ContractDefinition* existingContract = m_contracts.find(contract->name())->second.contract;
+          if (contract != existingContract)
```

Comment:
I swear that I know this.  I'm going to have to colorize my tabs so I don't keep doing this.